internal/webhook/general/eviction_webhook.go

```diff
@@ -0,0 +1,286 @@
+	// Parse eviction object
+	eviction, err := ew.parseEvictionObject(admissionReview.Request.Object.Raw)
+	if err != nil {
+		log.Error(err, "Failed to parse eviction object")
+
+		return &admissionv1.AdmissionResponse{
+			UID:     admissionReview.Request.UID,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  metav1.StatusFailure,
+				Message: fmt.Sprintf("Failed to parse eviction object: %v", err),
+				Code:    http.StatusBadRequest,
+			},
+		}
+	}
+
+	// Get pod information
+	pod, err := ew.getPodForEviction(eviction, admissionReview.Request.Namespace)
+	if err != nil {
+		log.Error(err, "Failed to get pod for eviction", "pod", eviction.Name)
+
+		return &admissionv1.AdmissionResponse{
+			UID:     admissionReview.Request.UID,
+			Allowed: false,
+			Result: &metav1.Status{
+				Status:  metav1.StatusFailure,
+				Message: fmt.Sprintf("Failed to get pod %s/%s: %v", admissionReview.Request.Namespace, eviction.Name, err),
+				Code:    http.StatusInternalServerError,
+			},
+		}
+	}
+
+	// Check if this is an Aerospike pod
+	if !ew.isAerospikePod(pod) {
+		log.V(1).Info("Allowing eviction of non-Aerospike pod", "pod", eviction.Name)
+		return nil // Allow eviction
+	}
+
+	// Block Aerospike pod eviction
+	log.Info("Blocking eviction of Aerospike pod", "pod", eviction.Name)
+
+	// Set annotation asynchronously (non-blocking)
+	// TODO: do we really want async here?
+	go ew.setEvictionBlockedAnnotationAsync(pod)
+
+	return &admissionv1.AdmissionResponse{
+		UID:     admissionReview.Request.UID,
+		Allowed: false,
+		Result: &metav1.Status{
+			Status: metav1.StatusFailure,
+			Message: fmt.Sprintf("Eviction of Aerospike pod %s/%s is blocked by admission webhook",
+				admissionReview.Request.Namespace, eviction.Name),
+			Reason: EvictionBlockedReason,
+			Code:   http.StatusForbidden,
+		},
+	}
+}
+
+// parseEvictionObject parses the eviction object from raw bytes
+func (ew *EvictionWebhook) parseEvictionObject(raw []byte) (*policyv1.Eviction, error) {
+	var eviction policyv1.Eviction
+	if err := json.Unmarshal(raw, &eviction); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal eviction object: %w", err)
+	}
+
+	return &eviction, nil
+}
+
+// getPodForEviction retrieves the pod that is being evicted
+func (ew *EvictionWebhook) getPodForEviction(eviction *policyv1.Eviction, namespace string) (*corev1.Pod, error) {
+	podKey := types.NamespacedName{
+		Name:      eviction.Name,
+		Namespace: namespace,
+	}
+
+	pod := &corev1.Pod{}
+
+	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+	defer cancel()
+
+	if err := ew.Client.Get(ctx, podKey, pod); err != nil {
+		return nil, fmt.Errorf("failed to get pod %s: %w", podKey.String(), err)
+	}
+
+	return pod, nil
+}
+
+// setEvictionBlockedAnnotationAsync sets the eviction blocked annotation asynchronously
+func (ew *EvictionWebhook) setEvictionBlockedAnnotationAsync(pod *corev1.Pod) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Second)
```

Comment:
We are defining these kind of timeout at multiple places. With this approach, eviction blocking will always not work in slow cluster (API server is running slow) as these requests will timeout even though they are within webhooks timeout range. We need to re-think the overall approach